pkg/api/server.go

```diff
@@ -69,6 +69,7 @@ func Start(uiBox embed.FS, loginUIBox embed.FS) {
 	txnManager := manager.GetInstance().TxnManager
 	pluginCache := manager.GetInstance().PluginCache
 	resolver := &Resolver{
+		scraperCache: manager.GetInstance().ScraperCache,
```

Comment:
`manager.singleton::RefreshScraperCache` creates a new instance of `scraper.Cache` and assigns it to a member on the singleton. The resolver's copy will likewise need to be refreshed.

Reply:
Solved this by adding a function to the resolver which reads it from the singleton. If the pointed-to object isn't stable, I'd rather have a single point where it needs to be updated than having multiple points you need to remember to update.

However, this then uncovers a pile-on problem: the singleton instance needs a mutex to protect access to the field as it's being updated, possibly by another goroutine. I'm not going to do that in this patch, however since it is a new problem and not one created by this PR.